libqtile/widget/redshift.py

```diff
@@ -0,0 +1,333 @@
+        ),
+        (
+            "brightness_fmt",
+            "Brightness: {brightness}",
+            "Text to display when showing brightness text"
+            "Available fields: 'brightness' brightness to set when redshift is enabled. ",
+        ),
+        (
+            "gamma_fmt",
+            "Gamma: {gamma_red}:{gamma_green}:{gamma_blue}",
+            "Text to display when showing gamma text"
+            "Available fields: 'gamma_red' gamma red value to set when redshift is enabled. "
+            "'gamma_blue' gamma blue value to set when redshift is enabled. "
+            "'gamma_green' gamma green value to set when redshift is enabled. ",
+        ),
+    ]
+
+    _dependencies = ["redshift"]
+
+    def __init__(self, **config):
+        _TextBox.__init__(self, **config)
+        self.add_defaults(Redshift.defaults)
+        self.add_defaults(base.PaddingMixin.defaults)
+        self.is_enabled = False
+        self._line_index = 0
+        self._lines = []
+        self.add_callbacks(
+            {
+                "Button1": self.click,
+                "Button4": self.scroll_up,
+                "Button5": self.scroll_down,
+            }
+        )
+
+    def _configure(self, qtile, bar):
+        _TextBox._configure(self, qtile, bar)
+        # disable redshift so we have a known initial state
+        self.reset_redshift()
+        self._set_lines()
+        self._set_text()
+
+    def draw(self):
+        """
+        Set the widget text and redraw the widget.
+        """
+        self.drawer.clear(self.background or self.bar.background)
+        self._set_text()
+        _TextBox.draw(self)
+
+    def format_object(self, obj):
+        """
+        Takes the given object and returns a formatted string representing the object.
+        """
+        if isinstance(obj, str):
+            return obj
+        else:
+            logger.warning(
+                "redshift: format_object: obj didn't match any of the instance types, returning empty string"
+            )
+            return ""
+
+    @expose_command
+    def scroll_up(self):
+        """
+        Scroll up to next item.
+        """
+        self._scroll(1)
+
+    @expose_command
+    def scroll_down(self):
+        """
+        Scroll down to next item.
+        """
+        self._scroll(-1)
+
+    def show_line(self):
+        """
+        Update the text with the the current index in lines.
+        """
+        if not self._lines:
+            return
+
+        obj = self._lines[self._line_index]
+
+        self.update(self.format_object(obj))
+
+    @expose_command
+    def enable_redshift(self):
+        """
+        Enable redshift with the temp, gamma and brightness params.
+        """
+        gamma_val = GammaGroup(self.gamma_red, self.gamma_green, self.gamma_blue)
+        try:
+            self._get_rs_driver().enable(self.temperature, gamma_val, self.brightness)
+        except Exception as e:
```

Comment:
```suggestion
        except subprocess.CalledProcessError as e:
```
It's bad practice to catch `Exception`. If some other error arises here then there's an unexpected bug which should be addressed rather than caught.